src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -679,8 +679,13 @@ public static void Beep()
         {
             if (!Console.IsOutputRedirected)
             {
-                ReadOnlySpan<byte> bell = "\u0007"u8; // Windows doesn't use terminfo, so the codepoint is hardcoded.
-                int errorCode = WindowsConsoleStream.WriteFileNative(OutputHandle, bell, useFileAPIs: Console.OutputEncoding.CodePage != UnicodeCodePage);
+                bool isUnicode = Console.OutputEncoding.CodePage == UnicodeCodePage;
```

Comment:
Are there other encodings that are not `UnicodeCodePage` and that may end up the bell characters into more than one byte? Would it make more sense to just encode the bell character using the `OutputEncoding` instead of trying to guess what the encoding may do here?

Reply:
You are right. Moreover, this method should never be on a hot path so we don't really gain anything by hardcoding the encoded bytes. I've changed the implementation, PTAL